core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -106,6 +106,9 @@
         Setting.byteSizeSetting("index.translog.flush_threshold_size", new ByteSizeValue(512, ByteSizeUnit.MB), Property.Dynamic,
             Property.IndexScope);
 
+    public static final Setting<TimeValue> INDEX_SEQ_NO_CHECKPOINT_SYNC_INTERVAL =
+        Setting.timeSetting("index.seq_no.checkpoint_sync_interval", new TimeValue(30, TimeUnit.SECONDS),
+            new TimeValue(-1, TimeUnit.MILLISECONDS), Property.Dynamic, Property.IndexScope);
```

Comment:
well, I think -1 is more intuitive than 0. In general we can maybe invest in setting parsing that accept -1 for disabling and requires a strictly positive value o.w. . This is however a different change IMO - here I just copied existing patterns from, for example, the refresh interval.